wagtail/admin/tests/test_whats_new.py

```diff
@@ -14,9 +14,11 @@ def setUpTestData(cls):
         cls.request_factory = RequestFactory()
         cls.user = cls.create_user(username="tester")
         cls.profile = UserProfile.get_for_user(cls.user)
-        cls.request = cls.request_factory.get("/")
-        cls.request.user = cls.user
-        cls.parent_context = {"request": cls.request}
+
+    def get_parent_context(self):
+        request = self.request_factory.get("/")
+        request.user = self.user
+        return {"request": request}
```

Comment:
Is it possible to use `setUp()` for this so that the rest of the file doesn't need to change?

Reply:
I don't think so... to clarify, the failure was that `test_render_html_user_dismissed` was rendering the panel content when it shouldn't. This happened even when running that test individually, which suggests that sharing the request object between test runs isn't the problem - rather, it's that the profile has already been retrieved and stashed on the request object before we've updated the database record. This means we have to ensure the database update happens before constructing the request - so constructing the request can't happen in `setUp`.

Reply:
Ah, got it, that makes sense. I wonder if it could also be solved by calling `self.user.refresh_from_db()` in the test, after calling `self.profile.save(...)`?

But anyway, I'm also happy with this approach!